lib/combinator/combine.js

```diff
@@ -67,29 +67,49 @@ Combine.prototype.run = function(sink, scheduler) {
 	return new CompoundDisposable(disposables);
 };
 
-function CombineSink(f, sinks, sink) {
+function CombineSink(f, sinks, scheduler, sink) {
 	this.f = f;
 	this.sinks = sinks;
+	this.scheduler = scheduler;
 	this.sink = sink;
-	this.ready = false;
+	this.active = false;
 	this.activeCount = sinks.length;
+	this.time = -Infinity;
 }
 
 CombineSink.prototype.event = function(t /*, indexSink */) {
-	if(!this.ready) {
-		this.ready = this.sinks.every(hasValue);
+	if(!this.active) {
+		this.active = this.sinks.every(hasValue);
 	}
 
-	if(this.ready) {
-		// TODO: Maybe cache values in their own array once this.ready
-		this.sink.event(t, invoke(this.f, map(getValue, this.sinks)));
+	if(this.active && t > this.time) {
+		this.time = t;
+		this.task = this.scheduler.at(t, new CombineTask(this.f, this.sinks, this.sink), 1);
```

Comment:
Hardcoding `1` here isn't a good solution.  There should be some constants, or maybe use `scheduler.defaultPriority + 1` or something like that instead.